packages/cli/docs/custom-plugins.md

```diff
@@ -0,0 +1,702 @@
+2. Add the `directory` property to the plugin options and use the plugin in you config file.
+
+```typescript
+// file-size.plugin.ts
+import { AuditOutput, AuditOutputs } from '@code-pushup/models';
+
+export type Options = {
+  // add the directory to the plugin options
+  directory: string;
+};
+
+const fileSizeAuditMeta: AuditOutput = {
+  slug: 'file-size',
+  title: 'File size',
+  // ...
+};
+
+async function runnerFunction(options: Options): Promise<AuditOutputs> {
+  return [
+    {
+      ...fileSizeAuditMeta,
+      value: 0,
+      // helper to for a nicer displayValue
+      displayValue: pluralizeToken('file', 0),
+      // We have always a score of 1 for now
+      score: 1,
+    } satisfies AuditOutput,
+  ];
+}
+```
+
+3. Get the raw data to perform the audit
+
+We need the raw data to create the `AuditOutput` and calculate the metrics.
+In our case we need to get the file name and size of all files in the provided directory.  
+The above code will recursively get the file size info for all files in the specified directory and its subdirectories.
+
+Let's create a runner function that uses the above code to get the file size data and turns it into the shape
+of `AuditOutputs`:
+
+**file-size `RunnerFunction`**
+
+```typescript
+// file-size.plugin.ts
+import { AuditOutput } from '@code-pushup/models';
+import { crawlFileSystem, pluralizeToken } from '@code-pushup/utils';
+import { RunnerFunction } from './plugin-config-runner';
+
+// ...
+
+function runnerFunction(options: Options): RunnerFunction {
+  return async (): Promise<AuditOutputs> => {
+    // use util to crawl the file system and transform each match into size infomation
+    const data = await crawlFileSystem(options, async filePath => {
+      const stats = await stat(filePath);
+      return { filePath, size: stats.size };
+    });
+
+    return [
+      {
+        ...fileSizeAuditMeta,
+        // number of crawled files
+        value: data.length,
+        // display helper for pluralisation
+        displayValue: pluralizeToken('file', data.length),
+        score: 0,
+      } satisfies AuditOutput,
+    ];
+  };
+}
+```
+
+Now we can execute the CLI with `npx code-pushup collect` and see a similar output as the following:
+
+<details>
+<summary> <b>stdout of CLI for the above code</b> (collapsed for brevity) </summary>
+
+```sh
+Code PushUp Report - @code-pushup/core@x.y.z
+
+File size plugin audits
+● File size audit                                                            2 files
+
+Categories
+┌──────────┬───────┬────────┐
+│ Category │ Score │ Audits │
+└──────────┴───────┴────────┘
+
+Made with ❤ by code-pushup.dev
+```
+
+</details>
+
+### RunnerConfig
+
+The second way to write a plugin runner is a `RunnerConfig`.
+This option is less flexible but can be used in cases when you have to use another CLI.
+why runner function can't be used...
```

Comment:
This isn't accurate, it's actually more flexible, but at the cost of more complexity.